src/platform/android/DeviceInstanceInfoProviderImpl.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "DeviceInstanceInfoProviderImpl.h"
+
+#include <platform/CHIPDeviceConfig.h>
+#include <platform/CHIPDeviceError.h>
+#include <platform/android/AndroidConfig.h>
+#include <platform/internal/GenericDeviceInstanceInfoProvider.ipp>
+
+namespace chip {
+namespace DeviceLayer {
+
+CHIP_ERROR DeviceInstanceInfoProviderImpl::GetHardwareVersionString(char * buf, size_t bufSize)
+{
+    CHIP_ERROR err;
+    size_t hardwareVersionLen = 0; // without counting null-terminator
+    err = Internal::AndroidConfig::ReadConfigValueStr(Internal::AndroidConfig::kConfigKey_HardwareVersionString, buf, bufSize,
+                                                      hardwareVersionLen);
+    if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
+    {
+        ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_DEFAULT_DEVICE_HARDWARE_VERSION_STRING), CHIP_ERROR_BUFFER_TOO_SMALL);
+        strcpy(buf, CHIP_DEVICE_CONFIG_DEFAULT_DEVICE_HARDWARE_VERSION_STRING);
+    }
+
+    return err;
+}
+
+CHIP_ERROR DeviceInstanceInfoProviderImpl::GetProductId(uint16_t & productId)
```

Comment:
@kkasperczyk-no HI Kamil ,in Android Platform, when you try to get the product ID, it cannot reached here, can you check that?

Reply:
@nicelyjust yes it looks like I did mistake when moving some methods to other modules. I pushed PR that I hope should fix this: https://github.com/project-chip/connectedhomeip/pull/20208